Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json

```diff
@@ -1,257 +1,256 @@
 {
-    "schemaVersion":"2.2",
-    "description":"Enable or Disable Pushing Agent Logs To CloudWatchLogs For Diagnostic Purposes",
-    "parameters": {
-      "LogLevel": {
-        "type": "String",
-        "description": "Log Level of SSM Agent",
-        "allowedValues": [
-          "Debug",
-          "Info",
-          "Error"
-        ]
+  "schemaVersion":"2.2",
+  "description":"Enable or Disable Pushing Agent Logs To CloudWatchLogs For Diagnostic Purposes",
+  "parameters": {
+    "LogLevel": {
+      "type": "String",
+      "description": "Log Level of SSM Agent",
+      "allowedValues": [
+        "Debug",
+        "Info",
+        "Error"
+      ]
+    },
+    "EnableCloudWatchUpload": {
+      "type": "String",
+      "description": "Enable Upload of Agent Logs To CloudWatch",
+      "allowedValues": [
+        "true",
+        "false"
+      ]
+    },
+    "LogGroup": {
+      "type": "String",
+      "default": "",
+      "description": "(Optional) Log Group Name. Will use the default group name if not specified. The log group would be created if not already present."
+    }
+  },
+  "mainSteps": [
+    {
+      "action":"aws:runPowerShellScript",
+      "precondition": {
+        "StringEquals": ["platformType", "Windows"]
       },
-      "EnableCloudWatchUpload": {
-        "type": "String",
-        "description": "Enable Upload of Agent Logs To CloudWatch",
-        "allowedValues": [
-          "true",
-          "false"
+      "name":"UploadSSMAgentLogsInWindows",
+      "inputs": {
+        "runCommand":[
```

Comment:
You should have the Powershell script as a separate file under your document folder, and provide a setup script to build it into the JSON document. It is hard to review a script when it is JSON escaped.